bug: Tests are failing if building without services-s3

### Describe the bug

If you build without services-s3, then the tests are failing in the lib documentation code:

```
failures:

---- src/lib.rs - (line 67) stdout ----
error[E0433]: failed to resolve: could not find `S3` in `services`
  --> src/lib.rs:77:33
   |
12 |     let mut builder = services::S3::default().bucket("test");
   |                                 ^^ could not find `S3` in `services`
```

### Steps to Reproduce

* Build opendal without any services turned on
* Run tests

### Expected Behavior

Tests should be working without any services enabled.

### Additional Context

_No response_

### Are you willing to submit a PR to fix this bug?

- [ ] Yes, I would like to submit a PR.

--- a/core/CONTRIBUTING.md
+++ b/core/CONTRIBUTING.md
@@ -5,7 +5,8 @@
 - `cargo check` to analyze the current package and report errors.
 - `cargo build` to compile the current package.
 - `cargo clippy` to catch common mistakes and improve code.
-- `cargo test` to run unit tests.
+- `cargo test --features tests` to run unit tests.
+- `cargo test` to run unit tests include doc tests.
 - `cargo bench` to run benchmark tests.
 
 Useful tips:
@@ -21,7 +22,13 @@ We have unit tests and behavior tests.
 
 Unit tests are placed under `src/tests`, organized by mod.
 
-To run all unit tests:
+To run unit tests:
+
+```shell
+cargo test --features tests
+```
+
+To run unit tests include doc tests:
 
 ```shell
 cargo test

